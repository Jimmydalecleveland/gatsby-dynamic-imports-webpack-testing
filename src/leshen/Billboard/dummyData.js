import lifestyle from '../../images/split-content-lifestyle.jpg'
import lifestyleFull from '../../images/billboard-lifestyle-full.jpg'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_FULL = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAGABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAdULJA//xAAZEAACAwEAAAAAAAAAAAAAAAABAgADBBL/2gAIAQEAAQUCTYTBa1kfXZ3/AP/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/AYf/xAAcEAACAQUBAAAAAAAAAAAAAAAAEQECEiExQXH/2gAIAQEABj8C0i6ZxTwlJeH/xAAZEAADAQEBAAAAAAAAAAAAAAABESEAQVH/2gAIAQEAAT8hN23nNcLWQCawQJjHn//aAAwDAQACAAMAAAAQBB//xAAWEQEBAQAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8QXJ//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQIBAT8QA2X/xAAaEAEAAgMBAAAAAAAAAAAAAAABACERMWGR/9oACAEBAAE/EGQY0bPrcwQjrko72algtuO3P//Z',
    src: lifestyleFull,
    srcSet: lifestyleFull,
    aspectRatio: 3.116883116883117,
    sizes: '(max-width: 800px) 100vw, 800px',
    presentationWidth: 2560,
  },
}
