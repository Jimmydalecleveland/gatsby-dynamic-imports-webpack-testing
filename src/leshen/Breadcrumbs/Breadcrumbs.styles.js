import styled from '@emotion/styled'
import { css } from '@emotion/core'
import Link from '../Link'

export const Breadcrumb = styled(Link)(
  ({ theme }) => css`
    padding-left: ${theme.spacing.xxs};
    text-decoration: none;

    &:link,
    :visited {
      color: ${theme.colors.dark.base.hex};
    }

    &:hover {
      text-decoration: none;
    }

    &::after {
      padding-left: ${theme.spacing.xxs};
      content: '/';
    }
  `
)

export const BreadcrumbContainer = styled.div(
  ({ theme }) =>
    css`
      top: ${theme.spacing.sm};
      position: absolute;
      z-index: 1;
      display: flex;
      font-size: ${theme.fonts.body.xs.size}px;
      text-transform: uppercase;

      @media (max-width: ${theme.breakpoints.sm}) {
        margin-top: -${theme.spacing.lg};
        position: relative;
        padding-bottom: ${theme.spacing.lg};
      }
    `
)

export const LastBreadcrumb = styled.p`
  font-size: ${({ theme }) => theme.fonts.body.xs.size}px;
  padding-left: ${({ theme }) => theme.spacing.xxs};
  line-height: unset;
  margin-top: 0;
  margin-bottom: 0;
  opacity: 0.6;
`
