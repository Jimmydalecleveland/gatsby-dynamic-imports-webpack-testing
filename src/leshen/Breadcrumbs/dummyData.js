export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
