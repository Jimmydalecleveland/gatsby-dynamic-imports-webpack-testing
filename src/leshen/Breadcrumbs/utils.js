/* eslint-disable import/prefer-default-export */

// This dictionary is to allow manual override for
// certain breadrumbs only when they are generated
// from a page's URL. It won't effect custom breadcrumbs.
const dictionary = {
  'a and e': 'A&E',
  'terms conditions': 'Terms & Conditions',
  e: 'E!',
}

export const checkDictionary = (word) =>
  dictionary[word.toLowerCase()] ? dictionary[word.toLowerCase()] : word
