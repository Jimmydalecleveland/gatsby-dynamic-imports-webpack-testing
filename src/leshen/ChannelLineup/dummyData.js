const CHANNEL_LINEUP_DATA = {
  channels: {
    edges: [
      {
        node: {
          name: 'History Channel',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'history-channel',
        },
      },
      {
        node: {
          name: 'Nickelodeon ',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'FX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fx',
        },
      },
      {
        node: {
          name: 'ESPN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'espn',
        },
      },
      {
        node: {
          name: 'FOX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fox',
        },
      },
      {
        node: {
          name: 'CNN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cnn',
        },
      },
      {
        node: {
          name: 'Cartoon Network',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cartoon-network',
        },
      },
      {
        node: {
          name: 'Comedy Central',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'comedy-central',
          url: '/tv/channels/comedy-central',
        },
      },
      {
        node: {
          name: 'Nickelodeon / Nick At Nite - West (NICKW)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'Turner Network Television (TNT)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'tnt',
          url: '/tv/channels/tnt',
        },
      },
      {
        node: {
          name: 'TV Land (TVLND)',
          packageAvailability: ['dn-americastop120+', 'dn-americastop250'],
          brandy_id: 'tvland',
        },
      },
      {
        node: {
          name: 'E! Entertainment Television (E!)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'e',
        },
      },
      {
        node: {
          name: 'Syfy Channel (SYFY)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'syfy',
        },
      },
      {
        node: {
          name: 'Eternal Word Television Network (EWTN)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'ewtn',
        },
      },
    ],
  },
  packages: {
    edges: [
      {
        node: {
          name: "America's Top 120",
          brandy_id: 'dn-americastop120',
        },
      },
      {
        node: {
          name: "America's Top 120+",
          brandy_id: 'dn-americastop120+',
        },
      },
      {
        node: {
          name: "America's Top 200",
          brandy_id: 'dn-americastop200',
        },
      },
      {
        node: {
          name: "America's Top 250",
          brandy_id: 'dn-americastop250',
        },
      },
    ],
  },
}

export default CHANNEL_LINEUP_DATA
