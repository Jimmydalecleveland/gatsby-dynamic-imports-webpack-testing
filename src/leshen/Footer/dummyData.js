export const FOOTER_NAV_ITEMS = [
  {
    text: 'Services',
    href: false,
    subnav: [
      {
        text: 'Clearlink Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'Bundles',
        href: '#',
      },
      {
        text: 'Frontier TV',
        href: '#',
      },
      {
        text: 'Dish TV',
        href: '#',
      },
      {
        text: 'Clearlink Phone',
        href: '#',
      },
      {
        text: 'Business',
        href: '#',
      },
    ],
  },
  {
    text: 'About',
    href: false,
    subnav: [
      {
        text: 'Store Locations',
        href: '#',
      },
      {
        text: 'Media Center',
        href: '#',
      },
      {
        text: 'Scholarship',
        href: '#',
      },
      {
        text: 'Contact Us',
        href: '#',
      },
    ],
  },
  {
    text: 'Español',
    href: false,
    subnav: [
      {
        text: 'Internet',
        href: '#',
      },
      {
        text: 'FiOS Internet',
        href: '#',
      },
      {
        text: 'DishLATINO',
        href: '#',
      },
      {
        text: 'Teléfono',
        href: '#',
      },
      {
        text: 'Paquetes',
        href: '#',
      },
    ],
  },
  {
    text: 'Terms',
    href: false,
    subnav: [
      {
        text: 'Legal',
        href: '#',
      },
      {
        text: 'Privacy Policy',
        href: '#',
      },
      {
        text: 'Terms of Service',
        href: '#',
      },
    ],
  },
]

export const DISCLAIMERS = [
  '©2019 Internet Service Partners. All Rights Reserved. The name CenturyLink and the pathways logo are trademarks of CenturyLink.',
  '*Geographic and service restrictions apply to AT&T services. Call to see if you qualify. Fiber optics may apply to all or part of the network, depending on your location.',
  '© 2019 AT&T Intellectual Property. AT&T, the AT&T logo and all other AT&T marks contained herein are trademarks of AT&T Intellectual Property and/or AT&T affiliated companies. All rights reserved. Subsidiaries and affiliates of AT&T Inc. provide products and services under the AT&T brand.',
]
