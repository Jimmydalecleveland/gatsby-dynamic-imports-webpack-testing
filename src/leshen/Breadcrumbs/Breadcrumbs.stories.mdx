import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Breadcrumbs, { ticketID } from '.'
import { CRUMB_DATA } from './dummyData'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|Breadcrumbs' component={Breadcrumbs} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Breadcrumbs

<Preview>
  <Story name="basic">
    <Breadcrumbs crumbs={CRUMB_DATA}/>
  </Story>
</Preview>

## Props
<Props of={Breadcrumbs} />

<Preview>
  <Story name="crumbless">
    <Breadcrumbs path="/tv/channels/a-and-e" />
  </Story>
</Preview>
