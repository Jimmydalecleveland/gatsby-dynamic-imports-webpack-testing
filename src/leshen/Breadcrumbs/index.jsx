import React from "react"
import PropTypes from "prop-types"
import Helmet from "react-helmet"

import * as Styled from "./Breadcrumbs.styles"
import { checkDictionary } from "./utils"
import { BREADCRUMB } from "../utils/dataTrackingValues"

const Breadcrumbs = ({
  crumbs,
  path,
  siteURL,
  removeTrailingSlashes,
  ...props
}) => {
  let crumbList = crumbs

  // Creates crumbs based off the page path if non have been provided
  if ((!crumbs || crumbs.length <= 0) && path !== "/") {
    crumbList = path.split("/").map((node, index) => {
      let currentPath =
        index === 0 ? "/" : `${path.substring(0, path.indexOf(node))}${node}`
      const shortName =
        index === 0 ? "Home" : checkDictionary(node.replace(/-/g, " "))
      if (!removeTrailingSlashes && index !== 0) {
        currentPath += "/"
      }
      return { shortName, path: currentPath }
    })
  }

  return (
    crumbList.length > 0 && (
      <Styled.BreadcrumbContainer {...props}>
        <Helmet>
          <script type="application/ld+json">
            {`
            {
              "@context":"https://schema.org",
              "@type": "BreadcrumbList",
                "itemListElement": [
                ${crumbList.map(
                  (crumb, i) =>
                    `
                      {
                        "@type": "ListItem",
                        "position": "${i + 1}",
                        "item": {
                          "@id": "${siteURL}${crumbList[i].path}",
                          "name": "${crumb.shortName}"
                        }
                      }
                    `
                )}
              ]
            }
          `}
          </script>
        </Helmet>
        {crumbList.map(({ shortName, path: crumbPath }, index) =>
          index === crumbList.length - 1 ? (
            <Styled.LastBreadcrumb key={shortName}>
              {shortName}
            </Styled.LastBreadcrumb>
          ) : (
            <Styled.Breadcrumb
              key={shortName}
              to={crumbPath}
              data-tracking={BREADCRUMB.LINK}
            >
              {shortName}
            </Styled.Breadcrumb>
          )
        )}
      </Styled.BreadcrumbContainer>
    )
  )
}

Breadcrumbs.defaultProps = {
  crumbs: [],
  path: "",
  siteURL: "",
  removeTrailingSlashes: false,
}

Breadcrumbs.propTypes = {
  crumbs: PropTypes.arrayOf(
    PropTypes.shape({
      shortName: PropTypes.string.isRequired,
      path: PropTypes.string.isRequired,
    })
  ),
  /** Current URL path. */
  path: PropTypes.string,
  /** Used in JSON-LD Schema.org data. */
  siteURL: PropTypes.string,
  removeTrailingSlashes: PropTypes.bool,
}

export default Breadcrumbs

export const ticketID = "CMS-157"
