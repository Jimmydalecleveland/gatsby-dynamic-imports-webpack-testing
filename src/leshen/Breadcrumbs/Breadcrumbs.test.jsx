import React from 'react'
import { render } from 'setup-tests'
import Breadcrumbs from '.'
import { CRUMB_DATA } from './dummyData'

describe('Breadcrumbs', () => {
  test('should render', () => {
    const { container, getByText } = render(<Breadcrumbs crumbs={CRUMB_DATA} />)
    expect(container).toMatchSnapshot()
    const internetLink = getByText(/internet/i)
    expect(internetLink).toHaveAttribute('href', '/internet')
    const gigPlansLink = getByText(/gigabit plans/i)
    expect(gigPlansLink).not.toHaveAttribute('href')
  })

  test('should render with path generated crumbs', () => {
    const { container, getByText } = render(
      <Breadcrumbs path="/tv/channels/a-and-e" removeTrailingSlashes />
    )
    const tvLink = getByText(/tv/i)
    expect(tvLink).toHaveAttribute('href', '/tv')
    const channelsLink = getByText(/channels/i)
    expect(channelsLink).toHaveAttribute('href', '/tv/channels')
    expect(container).toMatchSnapshot()
  })

  test('should render with trailing slashes', () => {
    const { getByText } = render(<Breadcrumbs path="/tv/channels/tnt" />)
    const tvLink = getByText(/tv/i)
    expect(tvLink).toHaveAttribute('href', '/tv/')
    const channelsLink = getByText(/channels/i)
    expect(channelsLink).toHaveAttribute('href', '/tv/channels/')
  })
})
