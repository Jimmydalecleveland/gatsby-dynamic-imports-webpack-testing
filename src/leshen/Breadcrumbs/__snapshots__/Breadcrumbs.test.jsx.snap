// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render 1`] = `
.emotion-8 {
  top: 16px;
  position: absolute;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  text-transform: uppercase;
}

@media (max-width:540px) {
  .emotion-8 {
    margin-top: -32px;
    position: relative;
    padding-bottom: 32px;
  }
}

.emotion-0 {
  padding-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:link,
.emotion-0:visited {
  color: #383d43;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0::after {
  padding-left: 8px;
  content: '/';
}

.emotion-6 {
  font-size: 12px;
  padding-left: 8px;
  line-height: unset;
  margin-top: 0;
  margin-bottom: 0;
  opacity: 0.6;
}

<div>
  <div
    class="emotion-8 emotion-9"
  >
    MOCK react-helmet
    <a
      aria-current="page"
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/"
    >
      Home
    </a>
    <a
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/internet"
    >
      Internet
    </a>
    <a
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/gigabit"
    >
      Gigabit
    </a>
    <p
      class="emotion-6 emotion-7"
    >
      Gigabit Plans
    </p>
  </div>
</div>
`;

exports[`Breadcrumbs should render with path generated crumbs 1`] = `
.emotion-8 {
  top: 16px;
  position: absolute;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  text-transform: uppercase;
}

@media (max-width:540px) {
  .emotion-8 {
    margin-top: -32px;
    position: relative;
    padding-bottom: 32px;
  }
}

.emotion-0 {
  padding-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:link,
.emotion-0:visited {
  color: #383d43;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0::after {
  padding-left: 8px;
  content: '/';
}

.emotion-6 {
  font-size: 12px;
  padding-left: 8px;
  line-height: unset;
  margin-top: 0;
  margin-bottom: 0;
  opacity: 0.6;
}

<div>
  <div
    class="emotion-8 emotion-9"
  >
    MOCK react-helmet
    <a
      aria-current="page"
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/"
    >
      Home
    </a>
    <a
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/tv"
    >
      tv
    </a>
    <a
      class="emotion-0 emotion-1"
      data-tracking="breadcrumb__link"
      href="/tv/channels"
    >
      channels
    </a>
    <p
      class="emotion-6 emotion-7"
    >
      A&E
    </p>
  </div>
</div>
`;
